### DISCUSSION TITLE
Are redisson collections "distributed" safe?

### DISCUSSION THREAD

**Discussion by oscarryz** [redisson/redisson#5576] [Q&A] [answered]
I can't get from the [documentation](https://github.com/redisson/redisson/wiki/7.-distributed-collections) if they actually are. 

For instance, if I get a queue, can two (or more) different JVM's read from it and get different values? 

e.g.  the following code running on different JVM's

```java
RQueue<String> queue = redis.getQueue(KEY);
queue.poll(); 
```

Would every JVM get a different value (granted the content of the queue are different) or do I need a to lock first? 

e.g. 

```java
RLock lock = redis.getLock(LOCK_KEY);
lock.lock();
try {
   RQueue<String> queue = redis.getQueue(KEY);
   queue.poll(); 
} finally {
    lock.unlock();
}
 ```

**Comment by mrniko** [comment, marked as answer]
All collection implementations are thread safe. You don't need to use locks

**Reply by oscarryz** [reply]
Thank you for the reply. 
Is it safe to assume they're thread safe across different JVM instances? Thread safe is usually used within the same JVM context, hence the question.
